pgcli/output.py

```diff
@@ -0,0 +1,53 @@
+import click
+
+# output interface:
+#   def emit(self, cmd: str, output: str) -> None
+
+
+class Pager:
+    """
+    An output target which emits to the pager via the PGCli object.
+    """
+
+    def __init__(self, cli):
+        self._cli = cli
+
+    def emit(self, cmd, output):
+        self._cli.echo_via_pager(output)
+
+
+class Stdout:
+    """
+    An output which emits to stdout via `click.echo()`.
+    """
+
+    def emit(self, cmd, output):
+        click.echo(output)
+
+
+class Log:
```

Comment:
So based on the `Pager/Stdout` comments I merged them into one class, since at this point there's no distinction  between them. I had a slight preference for more useful and specific classes that know how to output to pager or stdout, but I totally get the virtue of reducing knowledge of the `CLI` class.

`Log` I handled a little differently than your suggestion - the determination was effectively already being done based on the fact that the log file path was `Optional[str]`, so I made that the attribute of `Log` rather than a callable, it just seemed more straightforward this way.